jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildTarTask.java

```diff
@@ -119,7 +123,8 @@ public void buildTar()
       Path tarOutputPath = getTargetPath();
       PluginConfigurationProcessor pluginConfigurationProcessor =
           PluginConfigurationProcessor.processCommonConfigurationForTarImage(
-              new GradleRawConfiguration(jibExtension),
+              rawConfiguration,
+              containerizeWar,
```

Comment:
Can this value be obtained from project properties?

Reply:
If you mean just returning if the project is WAR (has the Gradle WAR Plugin applied or `<packaging>war` in `pom.xml`) and not about the final packaging decision after factoring in `packagingOverride`, the value can already be obtained from `GradleProjectProperties.getWarTask()` or directly from `MavenProject`. But not sure if you are referring to this.

So, this `containerizeWar` indicates the final packaging decision which should read the `packagingOverride` value, and knowing that, it may not fit very well into "project properties". But even if we decide to add a method like `getFinalPackagingType()` in project properties, it is not possible in the current design. A `GradleProjectProperites` instance is created via the static initializer `GradleProjectProperties.getForProject()`, and for it to create an instance, it has to know ahead if it should do WAR containerization or not (and the correct `appRoot` value based on the WAR containerization decision). In simple words, you can lay an egg only after having a chicken, but to have a chicken, you need an egg. In the past, I was wondering how our codebase can be changed to instantiate a `GradleProjectProperties` without requiring raw configuration, but haven't really found a good way. Currently, project properties is designed to provide `JavaLayerConfigurations`, and creating correct layers require processed config info.

Reply:
Ah, my confusion is because containerizeWar is sent to projectProperties, but that appears to be to do some calculations, not to hold the value.

Reply:
> Ah, my confusion is because containerizeWar is sent to projectProperties, but that appears to be to do some calculations, not to hold the value.

OK, I think you got what's happening now, but just in case, `Gradle/MavenProjectProperties` gets `containerizeWar` to actually do some calculations, for example, to compute the correct `JavaLayerConfigurations` which requires whether to know if it should do WAR containerization or not. It doesn't need to hold the `containerizeWar`.